src/shared/components/ColumnFilterDropDown.tsx

```diff
@@ -0,0 +1,44 @@
+import { Button, Input, Space } from 'antd';
+import { FilterDropdownProps } from 'antd/lib/table/interface';
+import React from 'react';
+
+export const ColumnFilterDropDown: (
+  props: FilterDropdownProps & { label: string }
```

Comment:
What relationship does this component have to antd? It looks to me like the fields are the same by coincidence rather than being exactly the same thing?

Reply:
Thanks for the review :)

To begin with, I have now removed these changes (apologies, the PR was in DRAFT state. I shouldn't have added reviewers).

That said, if we did upgrade antd the component `ColumnFilterDropDown` would not break (unless antd rename or remove `FilterDropdownProps` interface). Antd allows us to pass any custom react component as a value to its `filterDropdown` prop. As long as our custom component accepts the same (or an extension of) shape of props as `FilterDropdownProps` or its extension, it is correct from ts point of view.